### DISCUSSION TITLE
Lenis Scrolling with SSR in Next.js 15

### DISCUSSION THREAD

**Discussion by kerns** [darkroomengineering/lenis#412] [General]
I am wondering how to correctly integrate lenis + framer motion in a site I have based on [this starter](https://github.com/sanity-io/sanity-template-nextjs-clean), where I can't simply apply `"use client";` due to the data fetching model.

Is there a sane way to integrate lenis scrolling across the site, so that it doesn't need to be setup for each view? 🙏

**Comment by clementroche** [comment]
`lenis/react` doesn't require `'use client'` anymore since [v1.1.14](https://github.com/darkroomengineering/lenis/releases/tag/v1.1.14) unless you need to pass callback options such as `prevent`. In this case you can create a component that deal with Lenis on the client side and add it to your server page, take a look at our own implementation on [Satus](https://github.com/darkroomengineering/satus/blob/main/app/(pages)/(components)/lenis/index.jsx).

And framer-motion integration it really depends on your needs i guess but here is [how to use framer-motion ticker with lenis](https://github.com/darkroomengineering/lenis/blob/main/packages/react/README.md#framer-motion-integration) anyway.

**Comment by kerns** [comment]
Perfect... thank you!
